source/includes/usage-examples/write-code-examples.cpp

```diff
@@ -0,0 +1,96 @@
+#include <cstdint>
+#include <iostream>
+#include <vector>
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/uri.hpp>
+
+using bsoncxx::builder::basic::kvp;
+using bsoncxx::builder::basic::make_document;
+using bsoncxx::builder::basic::make_array;
+
+int main() {
+    mongocxx::instance instance{};
+    mongocxx::uri uri("<connection string>");
+    mongocxx::client client(uri);
+    auto db = client["db"];
+    auto collection = db["coll"];
+
+    {
+        // Inserts one document that stores the specified value
+        // start-insert-one
+        auto result = collection.insert_one(make_document(kvp("<field name>", <value>)));
+        // end-insert-one
+    }
+
+    {
+        // Inserts multiple documents that store the specified values
+        // start-insert-multiple
+        std::vector<bsoncxx::document::value> documents;
+        documents.push_back(make_document(kvp("<field name>", <value>)));
+        documents.push_back(make_document(kvp("<field name>", <value>)));
+
+        auto result = collection.insert_many(documents);
+        // end-insert-multiple
+    }
+
+    {
+        // Updates a document that matches the specified criteria
+        // start-update-one
+        auto query_filter = make_document(kvp("<field to match>", <value to match>));
+        auto update_doc = make_document(kvp("$set", make_document(kvp("<field name>", <value>))));
+
+        auto result = collection.update_one(query_filter.view(), update_doc.view());
+        // end-update-one
+    }
+
+    {
+        // Updates all documents that match the specified criteria
+        // start-update-multiple
+        auto query_filter = make_document(kvp("<field to match>", <value to match>));
```

Comment:
S: for the sake of the syntax highlighting,  make the second param a string (applies to all)
```suggestion
        auto query_filter = make_document(kvp("<field to match>", "<value to match>"));
```